pcdet/models/dense_heads/anchor_head_single.py

```diff
@@ -72,4 +73,10 @@ def forward(self, data_dict):
             data_dict['batch_box_preds'] = batch_box_preds
             data_dict['cls_preds_normalized'] = False
 
-        return data_dict
+            # not predicting type, predict first class for everything
+            B, num_anchors, C = batch_cls_preds.shape
+            type_preds = torch.zeros((B, num_anchors, self.num_class[1]), dtype=batch_cls_preds.dtype, device=batch_cls_preds.device)
```

Comment:
I'm not very convinced by the change of `self.num_class` to a list: I think it's dangerous and in the PR there's already a mistake caused by this change.

You could preserve `self.num_class` as an integer and introduce an additional parameter `num_aux_class` in `AnchorHeadWithType`.

Here you could just use `0` instead of `self.num_class[1]` to create the tensor `data_dict['batch_cls_type_preds']`.

Reply:
Yeah this was also bugging me, it was a quick solution but indeed a dangerous one, thanks for raising it, I've now added `self.num_aux_class` which is only passed to the anchor head.